object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,11 @@
+/* exported pick */
+function pick(source, keys) {
+  var obj = {};
+  for (var i = 0; i < keys.length; i++) {
+    obj[keys[i]] = source[keys[i]];
+    if (obj[keys[i]] === undefined) {
+      delete obj[keys[i]];
+    }
```

Comment:
You can tighten up this logic a bit. If `source[keys[i]]` is _not_ `undefined`, then assign it to `obj`

Reply:
Got ya, thanks Tim